src/library/Box/Log.php

```diff
@@ -80,6 +83,20 @@ public function __call($method, $params): void
                     }
 
                     break;
+            }
+             // Check if any of the keys in the params array match the keys in the maskKeys array
+             foreach ($params as $key => $value) {
+                if (is_array($value)) {
+                    foreach ($value as $k => $v) {
+                        if (in_array($k, $this->_maskKeys)) {
+                            $params[$key][$k] = '********';
+                        } else {
+                            $params[$key][$k] = $v;
+                        }
+                    }
+                } else if (in_array($key, $this->_maskKeys)) {
+                    $params[$key] = '********';
+                }
```

Comment:
This would arguably be much more robust if it was turned into a recursive function, then it would be able to search more than a single depth in the array. I'd set a recession limit of 10 or 15, though.

Additionally for as you've currently implemented it, we shouldn't be setting the variable when it's not masked as otherwise it's already correct. 
```DIFF
foreach ($value as $k => $v) {
                        if (in_array($k, $this->_maskKeys)) {
                            $params[$key][$k] = '********';
+                       }
-                        } else {
-                            $params[$key][$k] = $v;
-                        }
                    }
```

Reply:
I changed it and decided to throw out the values once the regression limit has been reached.

Reply:
I've already made this suggestion via Discord, but I'm going to share it here as well so it's not lost.

I'd modify the function to be like so:
```PHP
private function maskParams(array $params, int $depthLimit = 10): array|string
    {
        if ($depthLimit <= 0) {
            return 'Recursion limit reached while masking event log parameters'
        }

        foreach ($params as $key => $value) {
            if (is_array($value)) {
                $params[$key] = $this->maskParams($value, $depthLimit - 1);
            } elseif (in_array(strtolower($key), array_map('strtolower', $this->_maskKeys))) {
                $params[$key] = '********';
            }
        }

        return $params;
    }
 ```

This cleans up the limit logic a little bit and instead of throwing an exception (which would stop the logic flow, potentially breaking behavior unexpectedly) it simply drops any parameters that are past the depth of 10 and replaces it with a message stating that the recursion limit was reached.

I would also switch the `_maskKeys` to be private rather than protected as it just doesn't need to be accessed anywhere outside of the log class.